linera-service/src/proxy/grpc.rs

```diff
@@ -468,6 +471,44 @@ where
         Ok(Response::new(certificate.try_into()?))
     }
 
+    #[instrument(skip_all, err(Display))]
+    async fn download_certificates(
+        &self,
+        request: Request<CertificatesBatchRequest>,
+    ) -> Result<Response<CertificatesBatchResponse>, Status> {
+        let hashes = request
+            .into_inner()
+            .hashes
+            .into_iter()
+            .map(linera_base::crypto::CryptoHash::try_from)
+            .collect::<Result<Vec<linera_base::crypto::CryptoHash>, _>>()?;
+
+        // Use 70% of the max message size as a buffer capacity.
+        // Leave 30% as overhead.
+        let mut grpc_message_limit: GrpcMessageLimiter<linera_chain::data_types::Certificate> =
+            GrpcMessageLimiter::new((GRPC_MAX_MESSAGE_SIZE as f64 * 0.7) as usize);
```

Comment:
Maybe simpler and faster:
```suggestion
            GrpcMessageLimiter::new(GRPC_MAX_MESSAGE_SIZE * 7 / 10);
```

Reply:
Or should we make this a constant?
```suggestion
            GrpcMessageLimiter::new(GRPC_MAX_MESSAGE_SIZE as f64 * MESSAGE_SIZE_PERCENT / 100);
```

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/2644/commits/b8a8dcc8c51f3d70392922de9e42a94a7e2dc49a.